src/io/local_storage.ts

```diff
@@ -0,0 +1,227 @@
+import {IOHandler, ModelArtifacts, ModelArtifactsInfo, SaveResult} from './types';
+
+// tslint:enable:max-line-length
+
+const PATH_SEPARATOR = '/';
+const PATH_PREFIX = 'tensorflowjs_models';
+const INFO_SUFFIX = 'info';
+const MODEL_TOPOLOGY_SUFFIX = 'model_topology';
+const WEIGHT_SPECS_SUFFIX = 'weight_specs';
+const WEIGHT_DATA_SUFFIX = 'weight_data';
+
+/**
+ * Purge all tensorflow.js-saved model artifacts from local storage.
+ *
+ * @returns Paths of the models purged.
+ */
+export function purgeLocalStorageArtifacts(): string[] {
+  if (!(window && window.localStorage)) {
+    throw new Error(
+        'purgeLocalStorageModels() cannot proceed because local storage is ' +
+        'unavailable in the current environment.');
+  }
+  const LS = window.localStorage;
+  const purgedModelPaths: string[] = [];
+  for (let i = 0; i < LS.length; ++i) {
+    const key = LS.key(i);
+    const prefix = PATH_PREFIX + PATH_SEPARATOR;
+    if (key.startsWith(prefix) && key.length > prefix.length) {
+      const modelName = key.slice(prefix.length).split(PATH_PREFIX)[0];
+      if (purgedModelPaths.indexOf(modelName) === -1) {
+        purgedModelPaths.push(modelName);
+      }
+    }
+  }
+  return purgedModelPaths;
+}
+
+/**
+ * IOHandler subclass: Browser Local Storage.
+ *
+ * See the doc string to `browserLocalStorage` for more details.
+ */
+export class BrowserLocalStorage implements IOHandler {
+  protected readonly LS: Storage;
+  protected readonly modelPath: string;
+  protected readonly paths: {[key: string]: string};
+
+  constructor(modelPath: string) {
+    if (!(window && window.localStorage)) {
+      throw new Error(
+          'The current environment does not support local storage.');
+    }
+    this.LS = window.localStorage;
+
+    if (modelPath == null) {
+      throw new Error(
+          'For local storage, modelPath must not be null or undefined.');
+    }
+    if (!modelPath) {
+      throw new Error('For local storage, modelPath must not be empty.');
+    }
+    this.modelPath = modelPath;
+    const modelRoot = [PATH_PREFIX, this.modelPath].join(PATH_SEPARATOR);
+    this.paths = {
+      info: [modelRoot, INFO_SUFFIX].join(PATH_SEPARATOR),
+      topology: [modelRoot, MODEL_TOPOLOGY_SUFFIX].join(PATH_SEPARATOR),
+      weightSpecs: [modelRoot, WEIGHT_SPECS_SUFFIX].join(PATH_SEPARATOR),
+      weightData: [modelRoot, WEIGHT_DATA_SUFFIX].join(PATH_SEPARATOR),
+    };
+  }
+
+  /**
+   * Save model artifacts to browser local storage.
+   *
+   * See the documentation to `browserLocalStorage` for details on the saved
+   * artifacts.
+   *
+   * @param modelArtifacts The model artifacts to be stored.
+   * @returns An instance of SaveResult.
+   */
+  async save(modelArtifacts: ModelArtifacts): Promise<SaveResult> {
+    if (!(window && window.localStorage)) {
+      throw new Error(
+          'The current environment does not support local storage.');
```

Comment:
Do we want an io.io_util.enumerateSupportedHandlers? To be able to provide the list of support handlers in the error message?  In either case can we provide better details "LocalStorage requires a browser that supports XYZ, this program is not being run in a browser or not in a browser that supports that standard."